guides/release/templates/handlebars-basics.md

```diff
@@ -1,85 +1,203 @@
+For example, every Ember app has a file called `application.hbs`. You can write regular HTML markup there or in any other `hbs` file:
 
-```handlebars {data-filename=app/templates/application.hbs}
-Hello, <strong>{{this.firstName}} {{this.lastName}}</strong>!
+```hbs {data-filename=app/templates/application.hbs data-update=false}
+<h1>Starting simple</h1>
+<p>
+  This is regular html markup inside an hbs file
+</p>
+```
+
+If you make a mistake like forgetting to close a tag or missing a quotation mark, you will see some errors when you run the app with `ember serve`. However, reading the error message on the page or in your browser's developer console will get you going again right away.
+
+## Types of Templates
+
+There are two main types of Templates: Route Templates and Component Templates.
+
+A Route Template determines what is shown when someone visits a particular url, like `https://guides.emberjs.com/some-route`. A Component Template has bits of content that can be reused in multiple places throughout the app, like buttons or forms.
+
+The best way to tell if a Template is part of a Route or Component is to look at the filepath. If you look at an existing app, you will see Templates in many different places in the app folder structure! This is to help the app stay organized as it grows from one to one hundred Templates.
+
+## Making new Templates
+
+New templates should be made using the [Ember CLI](https://cli.emberjs.com). It helps ensure that the files go in the right place in the app folder structure, and that they follow the right file naming conventions. If you have never used the Ember CLI before, it is recommended to do the [Tutorial]().
+
+For example, either of these commands will generate `.hbs` Template files (and other things!) in your app:
+
+```sh
+ember generate component my-component-name
+ember generate route my-route-name
+```
+
+## Template restrictions
+
+A typical, modern web app is made of dozens of files that have to all be combined together into something the browser can understand. Ember does this work for you with zero configuration, but as a result, there are some rules to follow when it comes to adding assets into your HTML.
+
+You cannot use script tags directly within a template, and should use [actions]() or [Route Lifecycle Hooks]() to make your app responsive to user interactions and new data. If you are working with a non-Ember JavaScript library and need to use a `js` file from it, see the Guide section [Addons and Dependencies]().
+
+Similarly, you should not add links to CSS Stylesheets within the `hbs` file. Style rules should go in the `app/styles` directory instead. `app/styles/app.css` is included in your app's build by default. For CSS files within the styles directory, you can create multiple stylesheets and use regular CSS APIs like `import` to link them together. If you want to incorporate CSS from an npm package or similar, see [Addons and Dependencies]() for instructions.
```

Comment:
In MU/Octane layout:

`app/styles` ->  `src/ui/styles`
`app/styles/app.css` -> `src/ui/styles/app.css`